spark/src/test/scala/org/apache/comet/CometCastSuite.scala

```diff
@@ -895,6 +895,18 @@ class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  test("cast between decimals with different precision and scale") {
+    // cast between default Decimal(38, 18) to Decimal(9,1)
+    val values = Seq(BigDecimal("12345.6789"), BigDecimal("9876.5432"), BigDecimal("123.4567"))
+    val df = withNulls(values).toDF("a")
+    castTest(df, DataTypes.createDecimalType(7, 2))
```

Comment:
I think the original test case in the issue was casting to `Decimal(6, 2)` instead of `Decimal(7, 2)`?

Reply:
This is the result of the test with (6,2) - 

```scala
== Results ==
!== Correct Answer - 4 ==                  == Spark Answer - 4 ==
 struct<a:decimal(38,18),a:decimal(6,2)>   struct<a:decimal(38,18),a:decimal(6,2)>
 [null,null]                               [null,null]
 [123.456700000000000000,123.46]           [123.456700000000000000,123.46]
 [9876.543200000000000000,9876.54]         [9876.543200000000000000,9876.54]
![12345.678900000000000000,null]           [12345.678900000000000000,12345.68]
```

Reply:
The output is confusing here, but the left side is Spark and the right side is Comet. Spark is producing `null` but Comet is producing `12345.68`. We need to have Comet use the same logic as Spark here.